### ISSUE TITLE
`with` record update syntax removes other record fields

### ISSUE THREAD

**Issue by rockboynton** [dhall-lang/dhall-lang#1353]
See title. 

Example in dhall repl

```dhall
⊢ :let Config = { field1 : Optional { foo : Optional { bar : Optional Natural, baz : Natural } }, field2 : Natural }

⊢ :let defaultConfig : Config = { field1 = Some { foo = Some { bar = Some 69, baz = 42 } }, field2 = 0 }

⊢ defaultConfig with field1 = Some { foo = Some { bar = Some 2 } }

{ field1 = Some { foo = Some { bar = Some 2 } }, field2 = 0 }
```

As you can see, `baz` was removed, is this expected? 

```dhall
⊢ (defaultConfig with field1 = Some { foo = Some { bar = Some 2 } }) : Config

Error: Expression doesn't match annotation

{ field1 : …
           { foo : …
                   { - baz : …
                   , …
                   }
           }
, …
}

1│ (defaultConfig with field1 = Some { foo = Some { bar = Some 2 } }) : Config

(input):1:1
```

This makes it so I can't create a new Config from a default when I just want to change one field. Maybe there is a better way to do that?

**Comment by Gabriella439** [comment]
Yeah, there is a better way to do what you want.  I believe what you want is:

```dhall
⊢ defaultConfig with field1.?.foo.?.bar.? = 2

{ field1 = Some { foo = Some { bar = Some 2, baz = 42 } }, field2 = 0 }
````

**Comment by rockboynton** [comment]
@Gabriella439 Thank you for the prompt response!

I'm getting this error

```dhall
1 | defaultConfig with field1.?.foo.?.bar.? = 2
  |                           ^
unexpected '?'
expecting any label or whitespace
```

Is that a new feature? I can't seem to find it in the docs

```shell
[nix-shell:~]$ dhall --version
1.40.2
```

**Comment by mmhat** [comment]
@rockboynton The `.?` syntax got introduced in the [Dhall standard v22.0.0](https://github.com/dhall-lang/dhall-lang/releases/tag/v22.0.0) which is supported in the Dhall Haskell implementation since [1.41.0](https://hackage.haskell.org/package/dhall/changelog).
**TLDR;** Your need to update your Dhall interpreter to use this.

**Comment by rockboynton** [comment]
@mmhat Thank you, that does the trick. Would be nice to have the docs mention this feature as well.
